include/emp/prefab/ValueBox.hpp

```diff
@@ -0,0 +1,182 @@
+#ifndef EMP_VALUE_BOX_HPP
+#define EMP_VALUE_BOX_HPP
+
+#include <cmath>
+
+#include "../datastructs/set_utils.hpp"
+#include "../tools/string_utils.hpp"
+
+#include "../web/Div.hpp"
+#include "../web/Element.hpp"
+#include "../web/Input.hpp"
+
+#include "Collapse.hpp"
+#include "CommentBox.hpp"
+#include "FontAwesomeIcon.hpp"
+#include "ToggleSwitch.hpp"
+
+namespace emp::prefab {
+
+  const int min_slider_width = 100;
+
+  class ValueBox : public web::Div {
+
+    protected:
+    std::string box_base = this->GetID();
+
+    CommentBox description{emp::to_string(box_base, "_descr")};
+    Div view{emp::to_string(box_base, "_view")};
+
+    public:
+    ValueBox(
+      const std::string & label,
+      const std::string & desc,
+      const std::string & id=""
+    ) : Div(id), description() {
+      web::Element drop_button("button", emp::to_string(box_base, "_label"));
+      drop_button.AddAttr("class", "btn btn-link");
+
+      FontAwesomeIcon arrow_right("fa-angle-double-right");
+      arrow_right.AddAttr("class", "toggle_icon_right_margin");
+      FontAwesomeIcon arrow_up("fa-angle-double-up");
+      arrow_up.AddAttr("class", "toggle_icon_right_margin");
+      drop_button << arrow_right;
+      drop_button << arrow_up;
+      drop_button << label;
+
+      description.AddContent(desc);
+      CollapseCoupling(drop_button, description, false);
+      *this << drop_button;
+      *this << description;
+      *this << view;
+      this->AddAttr("class", "value_box");
+      view.AddAttr("class", "value_view");
+      description.AddAttr("class", "value_description");
+    }
+  };
+
+  class ValueDisplay : public ValueBox {
+
+    public:
+    ValueDisplay(
+      const std::string & label,
+      const std::string & value,
+      const std::string & desc,
+      const std::string & id=""
+    ) : ValueBox(label, desc, id) {
+      view << value;
+    }
+  };
+
+  class ValueControl : public ValueBox {
+    public:
+    web::Input mainCtrl;
+    ValueControl(
+      const std::string & label,
+      const std::string & desc,
+      const std::string & inital_value,
+      web::Input input,
+      const std::string & id=""
+    ) : ValueBox(label, desc, id), mainCtrl(input) {
+      view << mainCtrl;
+      mainCtrl.Value(inital_value);
+    }
+  };
+
+  class TextValueControl : public ValueControl {
+    public:
+    TextValueControl(
+      const std::string & label,
+      const std::string & desc,
+      const std::string & value,
+      const std::function<void(const std::string & val)> & onChange = [](std::string val) { ; },
```

Comment:
if you're not using a param you don't have to name it

i.e., `[](const std::string&){ ; },`